providers/implementations/signature/rsa_sig.c

```diff
@@ -1575,3 +1973,259 @@ const OSSL_DISPATCH ossl_rsa_signature_functions[] = {
+        { OSSL_FUNC_SIGNATURE_SIGN_MESSAGE_UPDATE,                      \
+          (void (*)(void))rsa_signverify_message_update },              \
+        { OSSL_FUNC_SIGNATURE_SIGN_MESSAGE_FINAL,                       \
+          (void (*)(void))rsa_sign_message_final },                     \
+        { OSSL_FUNC_SIGNATURE_VERIFY_INIT,                              \
+          (void (*)(void))rsa_##md##_verify_init },                     \
+        { OSSL_FUNC_SIGNATURE_VERIFY,                                   \
+          (void (*)(void))rsa_verify },                                 \
+        { OSSL_FUNC_SIGNATURE_VERIFY_MESSAGE_INIT,                      \
+          (void (*)(void))rsa_##md##_verify_message_init },             \
+        { OSSL_FUNC_SIGNATURE_VERIFY_MESSAGE_UPDATE,                    \
+          (void (*)(void))rsa_signverify_message_update },              \
+        { OSSL_FUNC_SIGNATURE_VERIFY_MESSAGE_FINAL,                     \
+          (void (*)(void))rsa_verify_message_final },                   \
+        { OSSL_FUNC_SIGNATURE_VERIFY_RECOVER_INIT,                      \
+          (void (*)(void))rsa_##md##_verify_recover_init },             \
+        { OSSL_FUNC_SIGNATURE_VERIFY_RECOVER,                           \
+          (void (*)(void))rsa_verify_recover },                         \
+        { OSSL_FUNC_SIGNATURE_FREECTX, (void (*)(void))rsa_freectx },   \
+        { OSSL_FUNC_SIGNATURE_DUPCTX, (void (*)(void))rsa_dupctx },     \
+        { OSSL_FUNC_SIGNATURE_QUERY_KEY_TYPES,                          \
+          (void (*)(void))rsa_sigalg_query_key_types },                 \
+        { OSSL_FUNC_SIGNATURE_GET_CTX_PARAMS,                           \
+          (void (*)(void))rsa_get_ctx_params },                         \
+        { OSSL_FUNC_SIGNATURE_GETTABLE_CTX_PARAMS,                      \
+          (void (*)(void))rsa_gettable_ctx_params },                    \
+        { OSSL_FUNC_SIGNATURE_SET_CTX_PARAMS,                           \
+          (void (*)(void))rsa_sigalg_set_ctx_params },                  \
+        { OSSL_FUNC_SIGNATURE_SETTABLE_CTX_PARAMS,                      \
+          (void (*)(void))rsa_sigalg_settable_ctx_params },             \
+        OSSL_DISPATCH_END                                               \
+    }
+
+#if !defined(OPENSSL_NO_RMD160) && !defined(FIPS_MODULE)
+IMPL_RSA_SIGALG(ripemd160, RIPEMD160);
+#endif
+IMPL_RSA_SIGALG(sha1, SHA1);
+IMPL_RSA_SIGALG(sha224, SHA2-224);
+IMPL_RSA_SIGALG(sha256, SHA2-256);
+IMPL_RSA_SIGALG(sha384, SHA2-384);
+IMPL_RSA_SIGALG(sha512, SHA2-512);
+IMPL_RSA_SIGALG(sha512_224, SHA2-512/224);
+IMPL_RSA_SIGALG(sha512_256, SHA2-512/256);
+IMPL_RSA_SIGALG(sha3_224, SHA3-224);
+IMPL_RSA_SIGALG(sha3_256, SHA3-256);
+IMPL_RSA_SIGALG(sha3_384, SHA3-384);
+IMPL_RSA_SIGALG(sha3_512, SHA3-512);
+#if !defined(OPENSSL_NO_SM3) && !defined(FIPS_MODULE)
+IMPL_RSA_SIGALG(sm3, SM3);
```

Comment:
I might have expected to see updates related to all these new RSA-* algs in doc/man7/EVP_PKEY-RSA.pod and doc/man7/EVP_SIGNATURE-RSA.pod

Reply:
Will add, but another day

Reply:
See 29d5467ad43091309227bdf6985179834f3bbfa3